### PR TITLE
GPC-492: Initial step function for GRO Ingestion

### PR DESCRIPTION
**Pull request by oskarwilliams**

### CODE REVIEW

File: lambdas/gro-split-file/src/main/resources/LepLambdaJsonLayout.json
Comment (ethanmills) [inline lambdas/gro-split-file/src/main/resources/LepLambdaJsonLayout.json:1]: I assume this has something to do with logging? Is it possible to change the filename to reflect this?

File: template.yaml [hunk 1]
```diff
@@ -499,3 +521,99 @@ Resources:
...
+  GroIngestionBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      # checkov:skip=CKV_AWS_21: We do not want to have versioning in this bucket as are treating it as a temporary store
+      AccessControl: Private
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: "aws:kms"
+              KMSMasterKeyID: !Ref MasterKmsKey
```
Comment (ethanmills) [inline template.yaml:534]:
```suggestion
              KMSMasterKeyID: !Ref MainKmsKey
```

File: template.yaml [hunk 2]
```diff
@@ -499,3 +521,99 @@ Resources:
...
+        TargetArn: !GetAtt AuthorisationDlq.Arn
+      Policies:
+        - Statement:
+            - Sid: KmsMasterKeyAccess
+              Effect: Allow
+              Action:
+                - 'kms:Decrypt'
+                - 'kms:GenerateDataKey'
+              Resource:
+                - !GetAtt MasterKmsKey.Arn
```
Comment (ethanmills) [inline template.yaml:615]:
```suggestion
                - !GetAtt MainKmsKey.Arn
```

File: template.yaml [hunk 3]
```diff
@@ -499,3 +521,99 @@ Resources:
...
+      Environment:
+        Variables:
+          POWERTOOLS_SERVICE_NAME: authorisation
+      AutoPublishAlias: latest
+      DeadLetterQueue:
+        Type: SQS
+        TargetArn: !GetAtt AuthorisationDlq.Arn
+      Policies:
+        - Statement:
+            - Sid: KmsMasterKeyAccess
```
Comment (ethanmills) [inline template.yaml:609]:
```suggestion
            - Sid: KmsMainKeyAccess
```

**Review by ethanmills — APPROVED** [review]
Some style points and PR interdependency, but nothing blocking

File: template.yaml [hunk 4]
```diff
@@ -119,6 +119,28 @@ Resources:
...
+  LogsBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      # checkov:skip=CKV_AWS_18: The logs bucket doesn't need logging.
+      AccessControl: Private
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: 'aws:kms'
+              KMSMasterKeyID: !Ref MasterKmsKey
```
[the remark this answers is not in the document]
Comment (oskarwilliams) [reply template.yaml:131]: Yep after rebase

File: template.yaml (hunk 1 above)
Comment (oskarwilliams) [reply template.yaml:534]: Ditto

File: template.yaml (hunk 2 above)
Comment (oskarwilliams) [reply template.yaml:615]: Ditto

File: template.yaml (hunk 3 above)
Comment (oskarwilliams) [reply template.yaml:609]: Ditto

File: lambdas/gro-split-file/src/main/resources/LepLambdaJsonLayout.json
Comment (oskarwilliams) [reply lambdas/gro-split-file/src/main/resources/LepLambdaJsonLayout.json:1]: Will do

**Pull request merged**